src/java.base/share/classes/java/io/PipedInputStream.java

```diff
@@ -444,10 +444,8 @@ else if (in > out)
      * @throws     IOException  {@inheritDoc}
      */
     @Override
-    public void close()  throws IOException {
+    public void close() throws IOException {
+        in = -1;
```

Comment:
I am not sure if this is write thing to do, variable 'in' is accessed from "synchronized void receive(byte[] b, int off, int len)". What if  'close()' is called simultaneously ?.

Reply:
It is definitely not the right thing to do - the synchronization is not "redundant" at all.